apps/namadillo/src/atoms/balance/functions.ts

```diff
@@ -56,3 +59,57 @@ export const getTotalDollar = (list?: TokenBalance[]): BigNumber | undefined =>
 export const getTotalNam = (list?: TokenBalance[]): BigNumber =>
   list?.find((i) => i.asset.base === namadaAsset.base)?.balance ??
   new BigNumber(0);
+
+const tnamAddressToDenomTrace = (
+  address: string,
+  tokenAddresses: (NativeToken | IbcToken)[]
+): DenomTrace | undefined => {
+  const token = tokenAddresses.find((entry) => entry.address === address);
+  const trace = token && "trace" in token ? token.trace : undefined;
+
+  if (typeof trace === "undefined") {
+    return undefined;
+  }
+
+  const separatorIndex = trace.lastIndexOf("/");
+
+  if (separatorIndex === -1) {
+    throw new Error(`Couldn't get denom trace from ${trace}`);
```

Comment:
isn't it better to return `undefined` here? I'm afraid of breaking the whole interface because we can't find a asset. We can use the `unknowAsset` for this case as well, just to render something

Reply:
Fixed!